shared/parcel.js

```diff
@@ -20,6 +20,10 @@ export function isParcel(asset) {
   )
 }
 
+export function isParcelId(id = '') {
+  return /^-?\d+,-?\d+$/.test(id)
```

Comment:
This makes me weary. We can have a new asset down the line that uses similar ids and the check will break.

To avoid this and the comment here https://github.com/decentraland/marketplace/pull/549/files#diff-59adc9d1d5a72a75a482f79a544fef9dR13 what do you think about adding the asset_type to the Routes?
Like this:

```jsx
<Route
  exact
  path={this.addLegacySupport(locations.buyParcel())}
  component={withAssetType(BuyParcelPage, ASSET_TYPE.parcel)}
/>
<Route
  exact
  path={this.addLegacySupport(locations.cancelSaleParcel())}
  component={withAssetType(CancelSaleEstatePage, ASSET_TYPE.parcel)}
/>
```

```js
function withAssetType(component, assetType) {
  return props => <Component assetType={assetType} {...props} />
}
```

or we can take a different approache, like using a `<RouteWithAssetType />`

```jsx
<RouteWithAssetType
  exact
  path={this.addLegacySupport(locations.cancelSaleParcel())}
  component={CancelSaleEstatePage}
  assetType=ASSET_TYPE.parcel}
/>
```

whatever you think is best

_code written on GitHub, use at your own risk_

Reply:
I used the `render` method of the `Route` component, because doing

```
function withAssetType(component, assetType) {
  return props => <Component assetType={assetType} {...props} />
}
```

that creates a new component on each render, which makes it to remount, bringing back the flickering and causing the internal state to be lost.